sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FilePartitionReaderFactory.scala

```diff
@@ -16,20 +16,23 @@
  */
 package org.apache.spark.sql.execution.datasources.v2
 
-import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.{FileSourceOptions, InternalRow}
 import org.apache.spark.sql.connector.read.{InputPartition, PartitionReader, PartitionReaderFactory}
 import org.apache.spark.sql.errors.QueryExecutionErrors
 import org.apache.spark.sql.execution.datasources.{FilePartition, PartitionedFile}
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
 abstract class FilePartitionReaderFactory extends PartitionReaderFactory {
+
+  protected def parsedOptions: FileSourceOptions
```

Comment:
For this new method, can we use simply `options` instead of `parsedOptions`?

Reply:
`parsedOptions` is the existing name, renaming it will need to change all the subclasses of `FilePartitionReaderFactory`. How about we do it in a separated PR?

Reply:
ah sorry, `parsedOptions` is not that widely used, let's rename it to `options` then.

Reply:
sure - done!